crates/core/src/mock/mock_starknet.rs

```diff
@@ -18,3 +74,60 @@ pub fn mock_starknet_provider(fixtures: Option<Vec<StarknetRpcFixture>>) -> Json
     }
     JsonRpcClient::new(transport)
 }
+
+#[cfg(test)]
+mod tests {
+    use reth_primitives::{Address, U256, U64};
+    use starknet::core::types::{BlockId, BlockTag};
+    use starknet_crypto::FieldElement;
+
+    use super::*;
+    use crate::client::client_api::KakarotEthApi;
+    use crate::client::config::StarknetConfig;
+    use crate::client::KakarotClient;
+
+    #[tokio::test]
+    async fn test_mock_provider_block_number() {
+        // Given
+        let fixtures = fixtures(vec![JsonRpcMethod::BlockNumber]);
+        let provider = mock_starknet_provider(Some(fixtures));
+
+        let config = StarknetConfig {
+            kakarot_address: FieldElement::from_hex_be(
```

Comment:
we can create a constant.rs file in mock directory, in order to group all our constants, and use lazy_static! like in client to initialize them